x/tally/types/gas_meter.go

```diff
@@ -37,8 +37,15 @@ func NewGasMeter(tallyGasLimit, execGasLimit, maxTallyGasLimit uint64, gasPrice
 		gasPrice:          gasPrice,
 	}
 
-	// Consume base gas cost, ignoring even if gas runs out.
-	gasMeter.ConsumeTallyGas(baseGasCost)
+	// For normal operations we first check if the gas limit is enough to cover
+	// the operation. For the base gas cost we always want to consume it as even
+	// running out of gas incurs a cost to the network.
+	if baseGasCost > gasMeter.tallyGasLimit {
+		gasMeter.tallyGasRemaining = 0
+	} else {
```

Comment:
Isn't this case taken care of by `ConsumeTallyGas()` logic anyways?

Reply:
Good catch, but the problem was that the gas should not be consumed in `ConsumeTallyGas()` as an operation that doesn't 'fit' in the remaining gas limit should not be executed.